File: src/main/java/org/foo/models/Company.java
package org.foo.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.foo.enums.CompanyStatus;
import org.hibernate.annotations.Where;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "companies")
@Where(clause = "is_deleted=false")
public class Company extends BaseEntity{

    @Column(unique = true)
    private String title;

    private String phone;

    private String website;

  private String addressLine1;

  private String addressLine2;


  private String city;


  private String state;


  private String country;


  private String zipCode;

    @Enumerated(EnumType.STRING)
    private CompanyStatus companyStatus;



}

File: lab1-frontend/src/app/common/company.ts
import {CompanyStatus} from "../enums/company-status";


export class CompanyDto {
  constructor(
    public id: number,
    public title: string,
    public phone: string,
    public website: string,
    public addressLine1: string,
    public addressLine2: string,
    public city: string,
    public state: string,
    public country: string,
    public zipCode: string,
    public companyStatus: CompanyStatus
  ) {}
}
